### PR TITLE
Feat: kvstore

### PR DESCRIPTION
**Pull request by jgryffindor**
Added KVStore neighborhood and endpoints support to events parser. 

## Description
**Client:**
- Added kvstore.put and kvstore.transfer to client methods and layouts.
- Added alert to transactions page for errors when network selector is on the wrong network context.
- Adding new charts for Network IPv4 bandwidth and packet counts.
- Added ExpandCode component to truncate content over 5KB and display an ellipsis to expand and option to collapse

**Server:**
- Added put/transfer to server event attributes parser.
- Improved analyzer to handle string addresses stored in the request arguments payload.
- Added update PUT endpoint to Prometheus query controller/service to make it easier to update queries. 

Fixes # (issue)

## Testing
- Tested locally against a clone of prod database. 

## Breaking Changes (if applicable)

None

### Put/Transfer Pages:
image
image

### ExpandCode Component:
**Expand:**
image

**Collapse:**
image

### New Metrics:
image

## Checklist:

- [ ] I have read and followed the CONTRIBUTING guidelines for this project.
- [ ] I have added or updated tests and they pass.
- [ ] I have added or updated documentation and it is accurate.
- [ ] I have noted any breaking changes in this module or downstream modules.

### CODE REVIEW

File: server/src/utils/protocol/attributes/kvstore/kvstore.put.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+export class KvstorePutAnalyzer extends Analyzer<ArgumentT, ResultT, EventT> {
+  static method = "kvstore.put";
+  static eventType = [7, 0];
+
+  parseArgs(sender: Address, payload: Map<any, any>): ArgumentT {
+    return {
+      key: payload.get(0),
+      value: payload.get(1),
+      owner: (parseAddress(payload.get(2)) || sender).toString(),
```
Comment (fmorency) [inline server/src/utils/protocol/attributes/kvstore/kvstore.put.ts:27]: I don't think that's right, see https://github.com/many-protocol/specification/blob/main/attributes/network/cddl/7_kvstore_commands.cddl

File: server/src/utils/protocol/attributes/kvstore/kvstore.transfer.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  EventT
+> {
+  static method = "kvstore.transfer";
+  static eventType = [13, 0];
+
+  parseArgs(sender: Address, payload: Map<any, any>): ArgumentT {
+    return {
+      key: payload.get(0),
+      owner: (parseAddress(payload.get(1)) || sender).toString(),
+      new_owner: parseAddress(payload.get(2)).toString(),
```
Comment (fmorency) [inline server/src/utils/protocol/attributes/kvstore/kvstore.transfer.ts:31]: I don't think that's right, see https://github.com/many-protocol/specification/blob/main/attributes/network/cddl/7_kvstore_commands.cddl

**Review by fmorency — CHANGES_REQUESTED** [review]
The `mnvqws6lhkm` `To` is not a MANY address. Something is wrong.

It would also be nice to display the `Key` and `Value` fields as well, in hex. Be careful as the `Value` field can be very big, e.g., 500Kb.

I glanced at the packet stuff but didn't do an in-depth review.

In the future, it would be nice if you could do 1 PR per feature as opposed to 1 PR with multiple unrelated changes.

File: server/src/utils/protocol/attributes/kvstore/kvstore.transfer.ts [hunk 3]
```diff
@@ -39,8 +39,8 @@ export class KvstoreTransferAnalyzer extends Analyzer<
   analyzeEvent(payload: Map<any, any>): EventT {
     return {
       key: payload.get(1).toString(),
```
[the remark this answers is not in the document]
Comment (jgryffindor) [reply server/src/utils/protocol/attributes/kvstore/kvstore.transfer.ts]: See update!

File: server/src/utils/protocol/attributes/kvstore/kvstore.transfer.ts [hunk 4]
```diff
@@ -26,9 +26,9 @@ export class KvstoreTransferAnalyzer extends Analyzer<
 
   parseArgs(sender: Address, payload: Map<any, any>): ArgumentT {
     return {
-      key: payload.get(0),
-      owner: (parseAddress(payload.get(1)) || sender).toString(),
-      new_owner: parseAddress(payload.get(2)).toString(),
+      key: parseBuffer(payload.get(0)).toString(),
```
[the remark this answers is not in the document]
Comment (jgryffindor) [reply server/src/utils/protocol/attributes/kvstore/kvstore.transfer.ts]: See update!

File: server/src/utils/protocol/attributes/kvstore/kvstore.put.ts [hunk 5]
```diff
@@ -22,9 +22,9 @@ export class KvstorePutAnalyzer extends Analyzer<ArgumentT, ResultT, EventT> {
 
   parseArgs(sender: Address, payload: Map<any, any>): ArgumentT {
     return {
-      key: payload.get(0),
-      value: payload.get(1),
-      owner: (parseAddress(payload.get(2)) || sender).toString(),
+      key: parseBuffer(payload.get(0)).toString(),
+      value: parseBuffer(payload.get(1)).toString(),
```
[the remark this answers is not in the document]
Comment (jgryffindor) [reply server/src/utils/protocol/attributes/kvstore/kvstore.put.ts]: See update!

File: server/src/utils/protocol/analyzer.ts [hunk 6]
```diff
@@ -40,6 +40,14 @@ export abstract class Analyzer<ArgumentT, ResponseT, EventT> {
...
+  if (value instanceof Uint8Array) {
+    return new Address(Buffer.from(value));
+  } else if (typeof value === "string") {
+    return Address.fromString(value);
+  }
+};
+
 export const tags = {
-  10000: (value: Uint8Array) => new Address(Buffer.from(value)),
+  10000: parseAddressTag,
```
[the remark this answers is not in the document]
Comment (jgryffindor) [reply server/src/utils/protocol/analyzer.ts:52]: Yahtzee!

**Review by fmorency — APPROVED** [review]

**Pull request merged**